src/User.php

```diff
@@ -438,11 +439,23 @@ public function cleanDBonPurge()
 
         $this->dropPictureFiles($this->fields['picture']);
 
-       // Ticket rules use various _users_id_*
+        // Ticket rules use various _users_id_*
         Rule::cleanForItemAction($this, '_users_id%');
         Rule::cleanForItemCriteria($this, '_users_id%');
 
-       // Alert does not extends CommonDBConnexity
+        // Users having the deleted user as supervisor
+        $rows = self::find([
+            'users_id_supervisor' => $this->fields['id']
+        ]);
+        $supervisedUser = new User();
+        foreach ($rows as $row) {
+            $supervisedUser->update([
+                'id' => $row['id'],
+                'users_id_supervisor' => 0,
+            ]);
+        }
```

Comment:
If I remember right, the best way to handle this is to add this relation in `inc/relation.constant.php`. With this, when user is purged, replacement user could be selected. It have to be tested, I am not 100% sure.

```diff
diff --git a/inc/relation.constant.php b/inc/relation.constant.php
index f20d34a399..900abdbc09 100644
--- a/inc/relation.constant.php
+++ b/inc/relation.constant.php
@@ -1410,6 +1410,7 @@ $RELATION = [
             'users_id_validate',
         ],
         '_glpi_useremails'              => 'users_id',
+        'glpi_users'                    => 'users_id_supervisor',
     ],
 
     'glpi_usertitles' => [
```

Reply:
It works. Refactored with aa7e22af3f